Fix Dockerfile for ubuntu:latest (16 xenial with python2.7 and 3.5)

aoblet opened the pull request.

aoblet committed:
Modify LibRaw build

- Remove from apt LibRaw download: OIIO and LibRaw1.17 are broken
LIBRAW_DECODER_FLATFIELD
- Compile LibRaw from scratch

aoblet committed:
Fix Dockerfile for ubuntu 16 xenial

- remove openctl (builded in dependencies)
- remove libboost version: take the latest
- change libtiff4 to 5
- install python-pip

aoblet committed:
Build dependencies: Refacto versions variables

aoblet committed:
Merge branch 'develop' of https://github.com/tuttleofx/TuttleOFX into docker

aoblet committed:
Build dependencies: expose first target versions

aoblet committed:
Dockerfile with python3.5 (default on ubuntu16 xenial)

aoblet commented:
I don't know how you prefer to manage pythonX configuration.

So there are two files:
- Dockerfile is configured for python2.7
- Dockerfile-python3 is configured for python3.5

cchampet commented on Dockerfile-python3:
```diff
@@ -0,0 +1,67 @@
+RUN cd ${TUTTLEOFX_DEV} && ${TUTTLEOFX_DEV}/tools/travis/build.sh
+
+RUN rm -rf ${TUTTLEOFX_BUILD} ${TRAVIS_BUILD_DIR}
+
```
Maybe we could provide a default command with CMD option.

```
CMD sam
```

https://docs.docker.com/engine/reference/builder/

See also ENTRYPOINT.

aoblet committed:
Move Dockerfiles to tools/docker

cchampet commented:
@aoblet Could you add in the Dockerfiles the instructions to clean the apt cache?

fabiencastan commented on tools/docker/python2.7/Dockerfile:
```diff
@@ -59,10 +59,9 @@ ENV PYTHON_VERSION=2.7 \
 RUN mkdir ${TUTTLEOFX_BUILD} ${TRAVIS_BUILD_DIR}
 
-COPY . ${TUTTLEOFX_DEV}
+RUN git clone https://github.com/tuttleofx/TuttleOFX.git ${TUTTLEOFX_DEV}
```
When you build the Dockerfile, the goal is to build the specific version you are using locally.

cchampet replied:
Ok so we need to find an other solution to get the source, because since the Dockerfile has been moved to "tools/docker/python2.7/", it is not possible to reference relative path out of the Docker wrokspace (ie. ../../..).

cchampet replied:
Ok see commit 0584067

aoblet committed:
Docker copy tuttle source code

Since Dockerfiles are wrapped into folders, the docker build command is:
[rootTuttle]$ docker build -f tools/docker/pythonX.Y/Dockerfile .
We cannot copy from parent folder context (COPY ../)

aoblet committed:
Dockerfiles optimize storage size

- clean apt cache
- Build and clean dependencies/TuttleOFX inside same RUN: avoid useless
storage stacking
- Gain 1.5GB size (3GB to 1.5GB)
